pkg/distro/fedora/images.go

```diff
@@ -329,12 +329,21 @@ func diskImage(workload workload.Workload,
 		return nil, err
 	}
 
+	d := t.arch.distro
+
 	img.Environment = t.environment
 	img.Workload = workload
 	img.Compression = t.compression
+
 	if bp.Minimal {
 		// Disable weak dependencies if the 'minimal' option is enabled
 		img.InstallWeakDeps = common.ToPtr(false)
+	} else {
+		// We only support disabling weak dependencies starting at Fedora 43, this is to
+		// ensure that we don't break older images when/if this toggle is added to an image.
+		if common.VersionGreaterThanOrEqual(d.osVersion, VERSION_WEAKDEPS) && t.disableWeakDeps {
+			img.InstallWeakDeps = common.ToPtr(false)
+		}
```

Comment:
I don't like these conditionals in the `ImageKind` generator function. Especially since this will become a problem once we consolidate the implementation across fedora and rhel.

I'd suggest to move this to the `ImageConfig` and instead set it in the image type's default `ImageConfig`.

Reply:
I've adjusted this PR to move `InstallWeakDeps` into the `ImageConfig` instead.